pymeasure/instruments/thyracont/smartline_v2.py

```diff
@@ -0,0 +1,485 @@
+        :return str: Response from the device after error checking.
+        """
+        self.write(f"{accessCode}{command}{compose_data(data)}")
+        self.wait_for(query_delay)
+        return self.read(command)
+
+    def read(self, command=None):
+        """Read from the device and do error checking.
+
+        :param str command: Original command sent to the device to compare it with the response.
+            None deactivates the check.
+        """
+        # Sometimes the answer contains 0x00 or values above 127, such that
+        # decoding fails.
+        response = self.read_bytes(-1, break_on_termchar=True)
+        if b"\x00" in response:
+            response = response.replace(b"\x00", b"")
+
+        # b"\r" is the termination character
+        got = response.rstrip(b"\r").decode('ascii', errors="ignore")
+        # Error checking
+        if got[3] == "7":
+            raise ConnectionError(self.errors[got[8:-1]])
+        if command is not None and got[4:6] != command:
+            raise ConnectionError(f"Wrong response to {command}: '{got}'.")
+        if calculate_checksum(got[:-1]) != got[-1]:
+            raise ConnectionError("Response checksum is wrong.")
+        return got[8:-1]
+
+    def check_set_errors(self):
+        """Check the errors after setting a property."""
+        self.read()
+        return []  # no error happened
+
+    " Main commands"
+
+    range = Instrument.measurement(
+        "0MR00", """Get the measurement range in mbar.""",
+        preprocess_reply=lambda r: r[1:], separator="L")
+
+    pressure = Instrument.measurement(
+        "0MV00", """Get the current pressure of the default sensor in mbar""",
+        preprocess_reply=lambda r: r.replace("UR", "0").replace("OR", "inf"),
+    )
+
+    # def Relays
+
+    display_unit = Instrument.control(
+        get_command="0DU00",
+        set_command="2DU%s",
+        docs="""Control the unit shown in the display. ('mbar', 'Torr', 'hPa')""",
+        values=['mbar', 'Torr', 'hPa'],
+        validator=validators.strict_discrete_set,
+        set_process=compose_data,
+        check_set_errors=True,
+    )
+
+    display_orientation = Instrument.control(
+        "0DO00", "2DO01%i",
+        """Control the orientation of the display in relation to the pipe ('top', 'bottom').""",
+        values={"top": 0, "bottom": 1},
+        map_values=True,
+        validator=validators.strict_discrete_set,
+        check_set_errors=True,
+    )
+
+    display_data = Instrument.control(
+        "0DD00", "2DD01%i",
+        """Control the display data source (strict SOURCES).""",
+        values=Sources,
+        cast=str_to_source,
+        validator=validators.strict_discrete_set,
+        check_set_errors=True,
+    )
+
+    def setHigh(self, high=""):
+        """Set the high pressure to `high` pressure in mbar."""
+        self.ask_manually(2, "AH", high)
+
+    def setLow(self, low=""):
```

Comment:
```suggestion
    def set_low(self, low=""):
```
Minor consistency change.